src/rgw/rgw_rados.cc

```diff
@@ -8944,33 +8944,55 @@ int RGWRados::cls_obj_usage_log_clear(const DoutPrefixProvider *dpp, string& oid
 }
 
 
-int RGWRados::remove_objs_from_index(const DoutPrefixProvider *dpp, RGWBucketInfo& bucket_info, list<rgw_obj_index_key>& oid_list)
+int RGWRados::remove_objs_from_index(const DoutPrefixProvider *dpp,
+				     RGWBucketInfo& bucket_info,
+				     const std::list<rgw_obj_index_key>& entry_key_list)
 {
   RGWSI_RADOS::Pool index_pool;
-  string dir_oid;
+  const auto& latest_log = bucket_info.layout.logs.back();
+  const rgw::bucket_index_layout_generation& current_index =
+    rgw::log_to_index_layout(latest_log);
+
+  const uint32_t num_shards = current_index.layout.normal.num_shards;
 
   uint8_t suggest_flag = (svc.zone->get_zone().log_data ? CEPH_RGW_DIR_SUGGEST_LOG_OP : 0);
+  std::map<int, std::string> index_oids;
 
-  int r = svc.bi_rados->open_bucket_index(dpp, bucket_info, &index_pool, &dir_oid);
-  if (r < 0)
+  int r = svc.bi_rados->open_bucket_index(dpp, bucket_info, std::nullopt,
```

Comment:
Is this the one you're talking about, @cbodley ?

Reply:
That call should open all shards and return the oid of each. So that call passes in a variable named `current_index` -- but that's the latest "indexing scheme", the last one in the log of "indexing schemes". The `std::nullopt` argument can be replaced with an int to specify a single shard to open.

Unless I'm misunderstanding what you're saying, @cbodley .

Reply:
i'm referring to the `open_bucket_index()` overload that this function used to call, which only returned a single `dir_oid`. i believe you removed its last call site here, so want to make sure nothing else misuses it in the future. i don't mind if that happens in a separate pr